packages/element/src/test/serialize.js

```diff
@@ -317,6 +317,36 @@ describe( 'renderElement()', () => {
 		expect( result ).toBe( 'inner provided|outer provided' );
 	} );
 
+	it( 'renders proper value through Context API when nested, distinct providers present', () => {
+		const {
+			Consumer: FirstConsumer,
+			Provider: FirstProvider,
+		} = createContext();
+		const {
+			Consumer: SecondConsumer,
+			Provider: SecondProvider,
+		} = createContext();
+
+		const result = renderElement(
```

Comment:
I see that e2e tests fail for blocks that have nesting. I'm wondering if there should be another unit tests that replicate how `InnerBlocks` works, if I follow it properly it would be something like:
```js
<FirstProvider value="First">
 	<FirstConsumer>
 		{ ( first ) => (
 			<SecondProvider value="Second">
 				<SecondConsumer>
 					{ ( second ) => `First: { first }, Second: ${ second }` }
 				</SecondConsumer>
			</SecondProvider>
 		) }
	</FirstConsumer>
</FirstProvider>
```

Although, I don't see any reason why it would make any difference.

Reply:
I'm not really sure what's going on with the end-to-end tests, to be honest. I've not been able to reproduce those same failures in my local environment.

Reply:
I added a test case for this in 90dacf058 for good measure. It passes, as expected. Still struggling with the end-to-end tests in Travis.

Reply:
I've just realized that we use one shared context for `InnerBlocks` so the following might be closer to the use case that fails on Travis:

```js
<Provider value="outer">
 	<Consumer>
 		{ ( outter ) => (
 			< Provider value="inner">
 				<Consumer>
 					{ ( inner ) => `Outer: { outer }, Inner: ${ inner }` }
 				</Consumer>
			</Provider>
 		) }
	</Consumer>
</Provider>
```

I don't think it's much different from a similar existing test though. I can't think of any reason why it fails on Travis :(

Reply:
Oh no :(